src/Type.hs

```diff
@@ -73,23 +80,39 @@ newtype Ty = Ty
   deriving (Eq, Show)
 makePrisms ''Ty
 
+tSyntax :: Ty
+tSyntax = Ty $ TyF TSyntax []
+
+tSignal :: Ty
+tSignal = Ty $ TyF TSignal []
+
+tString :: Ty
+tString = Ty $ TyF TString []
+
+tFun :: Ty -> Ty -> Ty
+tFun t1 t2 = Ty $ TyF TFun [t1, t2]
+infixr 9 `tFun`
+
+tMacro :: Ty -> Ty
+tMacro t = Ty $ TyF TMacro [t]
+
+tType :: Ty
+tType = Ty $ TyF TType []
+
+tDatatype :: Datatype -> [Ty] -> Ty
+tDatatype x ts = Ty $ TyF (TDatatype x) ts
+
+tSchemaVar :: Natural -> Ty
+tSchemaVar x = Ty $ TyF (TSchemaVar x) []
+
+tMetaVar :: MetaPtr -> Ty
+tMetaVar x = Ty $ TyF (TMetaVar x) []
+
 instance AlphaEq a => AlphaEq (TyF a) where
-  alphaCheck TSyntax TSyntax = pure ()
-  alphaCheck TSignal TSignal = pure ()
-  alphaCheck (TFun a1 b1) (TFun a2 b2) = do
-    alphaCheck a1 a2
-    alphaCheck b1 b2
-  alphaCheck (TMacro a) (TMacro b) =
-    alphaCheck a b
-  alphaCheck (TDatatype a args1) (TDatatype b args2) = do
-    guard (a == b)
+  alphaCheck (TyF ctor1 args1) (TyF ctor2 args2) = do
```

Comment:
I love short code. Especially code that lacks bugs like "forgot to compare TString"!